### PR TITLE
Adding PropertyFactory to plugins

### PR DESCRIPTION
**Pull request by vinaykumarchella**
Adding PropertyFactory to plugins so that any random properties can be accessed in plugins.

### CODE REVIEW

File: ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/cass/CassJavaDriverPlugin.java [hunk 1]
```diff
@@ -56,15 +59,24 @@
      * @throws Exception
      */
     @Override
-    public void init(DataGenerator dataGenerator) throws Exception {
-        Logger.info("Cassandra  Cluster: " + ClusterName);
+    public void init(DataGenerator dataGenerator, PropertyFactory propertyFactory) throws Exception {
         this.dataGenerator = dataGenerator;
+        this.propertyFactory = propertyFactory;
+
+        ClusterName = propertyFactory.getProperty("ndbench.config.cass.cluster").asString("localhost").get();
```
Comment (ipapapa) [inline ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/cass/CassJavaDriverPlugin.java]: cluster name to be `localhost`?

File: build.gradle [hunk 2]
```diff
@@ -48,7 +48,8 @@ subprojects {
         compile 'org.slf4j:slf4j-api:1.6.1'
         compile 'org.slf4j:slf4j-log4j12:1.6.1'
 
-
+        // Archaius-2
+        compile group: 'com.netflix.archaius', name: 'archaius2-core', version: '2.1.11'
```
Comment (ipapapa) [inline build.gradle]: why not `latest.release`?

File: ndbench-core/build.gradle [hunk 3]
```diff
@@ -6,13 +6,12 @@ dependencies {
...
     compile 'com.sun.jersey.contribs:jersey-guice:1.19.1'
-    
+
 
     // For getting all implementations of an Interface
     compile group: 'org.reflections', name: 'reflections', version: '0.9.10'
 
     // Archaius-2
-    compile group: 'com.netflix.archaius', name: 'archaius2-core', version: '2.1.11'
     compile group: 'com.netflix.archaius', name: 'archaius2-guice', version: '2.1.11'
```
Comment (ipapapa) [inline ndbench-core/build.gradle]: if you list the Archaius library under the `subprojects` or `repositories` clause of the main `build.gradle`, there is no reason to repeat it in subprojects.

File: ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/cass/CassJavaDriverPlugin.java (hunk 1 above)
Comment (vinaykumarchella) [inline ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/cass/CassJavaDriverPlugin.java]: `localhost` is the default name that C* gets when running locally.

File: ndbench-core/build.gradle (hunk 3 above)
Comment (vinaykumarchella) [inline ndbench-core/build.gradle]: Correct. That is why I removed `arhaius2-core` from subprojects and `archaius2-guice` in core project.

File: build.gradle (hunk 2 above)
Comment (vinaykumarchella) [inline build.gradle]: I think that is what I started with but gradle changed it to this version. I will give it a try again

**Pull request merged**